src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,117 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_INITIAL_ARRAY_CAPACITY = 10;
+    private Object[] values;
+    private int size;
+    private int capacityOfArray;
```

Comment:
why do we need `capacityOfArray` variable?

Reply:
changed with value.length